### DISCUSSION TITLE
Register-PnPManagementShellAccess AADSTS700016: Application with identifier '31359c7f-bd7e-475c-86db-fdb8c937548e' was not found in the directory 'XXX China'.

### DISCUSSION THREAD

**Discussion by Ruishuochina** [pnp/powershell#2877] [General]
Hi 

We're going to use PnP Management Shell on Azure Automation in 21 Venant in China, but we tried to use the PnPManagementShellAccess  cmdlet to register the PnP Management Shell  in Enterprise Application with the command Register-PnPManagementShellAccess -AzureEnvironment China -ShowConsentUrl -TenantName xxx.com.cn, and then we used the global admin to login the consetURL, meet the screenshot issues, anybody know how to fix it?

![image](https://user-images.githubusercontent.com/108163584/223658567-1ddfbe37-1738-4a32-a300-bba29061eb99.png)

**Comment by KnuckleHead007** [comment]
I have the same issue.

I am a Global Admin and I get the error trying:
Register-PnPManagementShellAccess -AzureEnvironment China -TenantName "[tenant identifier].partner.onmschina.cn"

And also trying (recommended from another thread):
Connect-PnPOnline -Url "https://[tenant identifier]-admin.sharepoint.cn" -PnPManagementShell -AzureEnvironment China -Tenant "[tenant identifier].partner.onmschina.cn"

**Comment by KnuckleHead007** [comment]
This worked. From:
https://pnp.github.io/powershell/articles/authentication.html

REGISTER
You are required to register your own Azure AD App in order to authentication.

Register-PnPAzureADApp -ApplicationName "PnP PowerShell" -Tenant [your tenant name].partner.onmschina.cn -Interactive -AzureEnvironment China -SharePointDelegatePermissions AllSites.FullControl -SharePointApplicationPermissions Sites.FullControl.All -GraphApplicationPermissions Group.ReadWrite.All -GraphDelegatePermissions Group.ReadWrite.All

Use your Global Admin creds.

CONNECT
Connect-PnPOnline -Credential [account PS Creds] -Url [yoursite] -AzureEnvironment China -ClientId "[client/app id obtained above]"

**Reply by cristian-agresta-dasit** [reply]
Thanks for the link! It worked also for me. BUT.
It created also the PnP PowerShell Certificate. To use it for a connection in service mode (no interaction available) I need a password... but which one? It did not ask me a password when I create it!
